### PR TITLE
Get detox in workflows

### PR DESCRIPTION
**Pull request by efalkner**
WIP

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -34,3 +34,65 @@ jobs:
       - run: yarn test
         env:
           CI: true
+
+  build-ios:
```
Comment (tstirrat) [inline .github/workflows/test.yml:38]: we already have build-ios, and similar job names can confuse GH, make this e2e-ios

File: e2e/pages/EnableLocation.po.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const buttonlabel = 'Enable Location';
+const screenshotText = 'Enable Location Page';
+const screenShotWithMenuText = 'Location Permissions Dialog';
+
+class EnableLocation {
+  tapButton = async () => {
```
Comment (tstirrat) [inline e2e/pages/EnableLocation.po.js:6]: these don't need `this` binding so normal form is fine

```js
async tapButton() {}
```

File: ios/Podfile [hunk 3]
```diff
@@ -3,6 +3,8 @@ require_relative '../node_modules/@react-native-community/cli-platform-ios/nativ
 
 target 'COVIDSafePaths' do
   # Pods for COVIDSafePaths
+  project 'COVIDSafePaths.xcodeproj'
```
Comment (tstirrat) [inline ios/Podfile:6]: only needed if you have old PrivateKit artifacts under `ios/`

But probably work committing.

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (tstirrat) [inline .vscode/settings.json:1]: remove this file

File: app/components/Button.js [hunk 5]
```diff
@@ -36,7 +36,10 @@ class Button extends React.Component {
             buttonStyle ? buttonStyle : styles.container,
             { height: buttonHeight },
           ]}
-          onPress={onPress}>
+          onPress={onPress}
+          accessible
```
Comment (tstirrat) [inline app/components/Button.js:40]: is `accessible` needed if the label is provided?

File: .eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
+module.exports = {
```
Comment (tstirrat) [inline .eslintrc.js:1]: if you can get this to be marked as a file move that would be helpful to preserve blame/history :)

File: ios/Podfile (hunk 3 above)
Comment (efalkner) [reply ios/Podfile:6]: This is from a merge, not intentional commit on my part

File: app/components/Button.js (hunk 5 above)
Comment (efalkner) [reply app/components/Button.js:40]: from a merge

File: .eslintrc.js (hunk 6 above)
Comment (efalkner) [reply .eslintrc.js:1]: will look into how

File: e2e/pages/EnableLocation.po.js (hunk 2 above)
Comment (efalkner) [reply e2e/pages/EnableLocation.po.js:6]: Yup, cruft from a previous approach. will remove

File: .eslintrc.js (hunk 6 above)
Comment (efalkner) [reply .eslintrc.js:1]: Spoke with Tim about this offline, we determined it wasn't a blocker

File: ios/Podfile (hunk 3 above)
Comment (tstirrat) [reply ios/Podfile:6]: its fine to commit. it helps when people have the old PrivateKit files left in their workdir.

File: ios/Podfile.lock [hunk 7]
```diff
@@ -23,9 +23,9 @@ PODS:
     - DoubleConversion
     - glog
   - glog (0.3.5)
-  - Permission-LocationAlways (2.0.10):
+  - Permission-LocationAlways (2.1.2):
```
Comment (tstirrat) [inline ios/Podfile.lock:26]: Are these pod updates needed?

File: package.json [hunk 8]
```diff
@@ -43,6 +47,7 @@
     "@testing-library/react-native": "^5.0.3",
     "dayjs": "^1.8.24",
     "emotion-theming": "^10.0.27",
+    "eslint-plugin-detox": "^1.0.0",
```
Comment (tstirrat) [inline package.json:50]: duplicated in dev and deps

File: ios/Podfile.lock (hunk 7 above)
Comment (efalkner) [reply ios/Podfile.lock:26]: Just a result of the merge.

File: package.json (hunk 8 above)
Comment (efalkner) [reply package.json:51]: Should only be in dev. Will update.

File: .vscode/settings.json (hunk 4 above)
Comment (tstirrat) [reply .vscode/settings.json:1]: Revert this file and then I'm happy and ready to merge!

File: e2e/pages/Onboarding3.po.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+const buttonlabel = 'Next';
+const screenText =
+  'If you test positive, you can choose to donate your data anonymously';
+const screenshotText = 'Third Onboarding Page';
```
Comment (tstirrat) [inline e2e/pages/Onboarding3.po.js:4]: nit: these come out unordered in filesystem, perhaps `Onboarding - Page 3` and so on

**Pull request merged**